src/main/java/org/opensearch/security/ssl/OpenSearchSecuritySSLPlugin.java

```diff
@@ -664,4 +666,29 @@ public List<String> getSettingsFilter() {
         settingsFilter.add("plugins.security.*");
         return settingsFilter;
     }
+
+    @Override
+    public Optional<SecureSettingsFactory> getSecureSettingFactory(Settings settings) {
+        return Optional.of(new OpenSearchSecureSettingsFactory(settings, sks, NOOP_SSL_EXCEPTION_HANDLER));
+    }
+
+    protected Settings migrateSettings(Settings settings) {
+        final Settings.Builder builder = Settings.builder().put(settings);
+
+        builder.remove(NetworkModule.TRANSPORT_SSL_DUAL_MODE_ENABLED_KEY);
+        builder.remove(NetworkModule.TRANSPORT_SSL_ENFORCE_HOSTNAME_VERIFICATION_RESOLVE_HOST_NAME_KEY);
+        builder.remove(NetworkModule.TRANSPORT_SSL_ENFORCE_HOSTNAME_VERIFICATION_KEY);
+
+        builder.put(NetworkModule.TRANSPORT_SSL_DUAL_MODE_ENABLED_KEY, SecuritySettings.SSL_DUAL_MODE_SETTING.get(settings));
+        builder.put(
+            NetworkModule.TRANSPORT_SSL_ENFORCE_HOSTNAME_VERIFICATION_RESOLVE_HOST_NAME_KEY,
```

Comment:
Isn't this setting from the network module the 'new setting' so it should take precedent over the existing setting from the security plugin?  Alternatively only allowing this setting defined in one place seems like a way to avoid unclear defaults, what do you think?

Reply:
> Isn't this setting from the network module the 'new setting' so it should take precedent over the existing setting from the security plugin?

Very valid question, at this moment the new setting is not used - the configuration is purely driven by security plugin settings. However the end solution I see would be this: the `security` plugin uses the secure transport as is (this method would be gone), but the we may need to add fallbacks for  `NetworkModule.*` to use `security` plugin settings (I am working towards that in https://github.com/opensearch-project/OpenSearch/issues/12903)

Reply:
What do you think about throwing an error if the `NetworkModule` is set in the mean time, and we circle back on this when considering pushing more settings into core?  I'd hate for an upgrade to suddenly cause a behavior change after a feature came out of an incubation cycle.

Reply:
Sure, I will work through it, thanks!

Reply:
@peternied addressed, curious what do you think, thank you!

Reply:
Nice - could you also include a unit test to validate?

Reply:
@peternied done! thank you!